kratos/geometries/triangle_2d_3.h

```diff
@@ -721,6 +721,8 @@ template<class TPointType> class Triangle2D3
 
         rResult = ZeroVector(3);
 
+        const double tolerance = 1e-14;
```

Comment:
only thing is that i would make this tolerance much smaller, say

     constexpr double tolerance = 1000.0*std::numeric_limits...

Reply:
I agree with @RiccardoRossi to use very small tolerance, even the numerical limit itself using an alternative direction for calculating the local coordinates....

Reply:
@pooyan-dadvand I've just implemented it in another way to avoid the division by 0. Give me a few minutes to commit the changes.